vortex-array/src/compute/boolean.rs

```diff
@@ -40,12 +40,7 @@ where
         rhs: &ArrayData,
         op: BinaryOperator,
     ) -> VortexResult<Option<ArrayData>> {
-        let array_ref = <&E::Array>::try_from(lhs)?;
-        let encoding = lhs
-            .encoding()
-            .as_any()
-            .downcast_ref::<E>()
-            .ok_or_else(|| vortex_err!("Mismatched encoding"))?;
+        let (array_ref, encoding) = downcast_array_ref::<E>(lhs)?;
```

Comment:
just small general note - can we make the function be `downcast`/`downcast_ref` and part of the `ArrayData` impl so its just:
```rust
let (array_ref, encoding) = lhs.downcast_ref::<E>()?;
```

Reply:
My (possibly fallible) notes indicate @gatesn requested this name instead of `downcast_ref`.

Reply:
And Done on the second part (make it a method).